modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java

```diff
@@ -214,83 +222,120 @@ private void initializeAttributeEvaluators(AttributeEvaluator ... attributeEvalu
-                                     }
-                                 }
-
-                                 private MuleEvent resetMuleEventForNewThread(MuleEvent event)
-                                 {
-                                     // Reset access control for new thread
-                                     ((DefaultMuleEvent)event).resetAccessControl();
-                                     // Set RequestContext ThreadLocal in new thread for backwards compatibility
-                                     OptimizedRequestContext.unsafeSetEvent(event);
-                                     return event;
-                                 }
-                             }, getWorkManager(muleEvent));
+        getHttpClient().send(httpRequest, resolveResponseTimeout(muleEvent),
+                             followRedirects.resolveBooleanValue(muleEvent),
+                             resolveAuthentication(muleEvent),
+                             createNonBlockingCompletionHandler(muleEvent, originalCompletionHandler, retryCount, authentication, httpRequest),
+                             getWorkManager(muleEvent));
+    }
+
+    private CompletionHandler<HttpResponse, Exception> createNonBlockingCompletionHandler(final MuleEvent muleEvent,
+                                                                                          final CompletionHandler originalCompletionHandler,
+                                                                                          final int retryCount,
+                                                                                          final HttpAuthentication authentication,
+                                                                                          final HttpRequest httpRequest)
+    {
+        return new CompletionHandler<HttpResponse, Exception>()
+        {
+            @Override
+            public void onFailure(Exception exception)
+            {
+                MessagingException msgException = new MessagingException(CoreMessages.createStaticMessage(getErrorMessage(httpRequest)),
+                                                                         resetMuleEventForNewThread(muleEvent),
+                                                                         exception,
+                                                                         DefaultHttpRequester.this);
+                checkIfRemotelyClosed(exception);
+                if(shouldRetryRemotelyClosed(exception, retryCount))
+                {
+                    try
+                    {
+                        innerProcessNonBlocking(muleEvent, originalCompletionHandler, retryCount - 1);
+                    }
+                    catch (MuleException e)
+                    {
+                        originalCompletionHandler.onFailure(new MessagingException(CoreMessages.createStaticMessage("Error sending HTTP request"), muleEvent, e));
```

Comment:
Same comment as before, getErrorMessage will show the request that failed which was an ER to improve troubleshooting.